util/mmio.c

```diff
@@ -6,6 +6,81 @@
 #include <pthread.h>
 #include <stdbool.h>
 
+#ifdef __s390x__
+#include <sys/auxv.h>
+#include <ccan/minmax.h>
+
+bool s390_is_mio_supported;
+
+static __attribute__((constructor)) void check_mio_supported(void)
+{
+#ifdef HWCAP_S390_PCI_MIO
```

Comment:
I'd rather you use the usual shim mechanism for this constant to ensure it is always declared regardless of the libc version. ie have cmake test it and replace sys/auxv.h like we do for other things

Reply:
Ok, thanks, will look into that.

Reply:
In v6 that I just pushed, I added a `buildlib/fixup-include/sys-auxv.h` with the `HWCAP_S390_PCI_MIO`. We'll likely have backports for glibc 2.34 for this flag in the major distributions with s390 support anyway but in the code the shim method is definitely nicer than the inline `#ifdef`.